pact4s/src/test/scala/org/broadinstitute/dsde/rawls/provider/RawlsProviderSpec.scala

```diff
@@ -0,0 +1,322 @@
+        mockSnapshotServiceConstructor(RawlsRequestContext(userInfo = mockUserInfo, otelContext = mockOtelContext)),
+        dataRepoSnapshotResource
+      )
+    case _ =>
+      loggerIO.debug("State not found")
+  }
+
+  private def mockEnumerateSnapshots(mockSnapshotService: SnapshotService,
+                                     mockResponse: SnapshotListResponse
+  ): OngoingStubbing[Future[SnapshotListResponse]] =
+    when {
+      mockSnapshotService.enumerateSnapshotsById(anyString(), anyInt(), anyInt(), any[Option[UUID]])
+    } thenReturn {
+      Future.successful(mockResponse)
+    }
+
+  private def mockCreateSnapshot(mockSnapshotService: SnapshotService,
+                                 mockResponse: DataRepoSnapshotResource
+  ): OngoingStubbing[Future[DataRepoSnapshotResource]] =
+    when {
+      mockSnapshotService.createSnapshotByWorkspaceId(anyString(), any[NamedDataRepoSnapshot])
+    } thenReturn {
+      Future.successful(mockResponse)
+    }
+
+  lazy val pactBrokerUrl: String =
+    sys.env.getOrElse("PACT_BROKER_URL", "")
+  lazy val pactBrokerUser: String = sys.env.getOrElse("PACT_BROKER_USERNAME", "")
+  lazy val pactBrokerPass: String = sys.env.getOrElse("PACT_BROKER_PASSWORD", "")
+  // Provider branch, semver
+  lazy val providerBranch: String = sys.env.getOrElse("PROVIDER_BRANCH", "")
+  lazy val providerVer: String = sys.env.getOrElse("PROVIDER_VERSION", "")
+  // Consumer name, branch, semver (used for webhook events only)
+  lazy val consumerName: Option[String] = sys.env.get("CONSUMER_NAME")
+  lazy val consumerBranch: Option[String] = sys.env.get("CONSUMER_BRANCH")
+  // This matches the latest commit of the consumer branch that triggered the webhook event
+  lazy val consumerVer: Option[String] = sys.env.get("CONSUMER_VERSION")
+
+  var consumerVersionSelectors: ConsumerVersionSelectors = ConsumerVersionSelectors()
+  // consumerVersionSelectors = consumerVersionSelectors.mainBranch
+  // The following match condition basically says
+  // 1. If verification is triggered by consumer pact change, verify only the changed pact.
+  // 2. For normal Rawls PR, verify all consumer pacts in Pact Broker labelled with a deployed environment (alpha, dev, prod, staging).
+  consumerBranch match {
+    case Some(s) if !s.isBlank => consumerVersionSelectors = consumerVersionSelectors.branch(s, consumerName)
+    case _                     => consumerVersionSelectors = consumerVersionSelectors.deployedOrReleased.mainBranch
+  }
+
+  val provider: ProviderInfoBuilder =
+    ProviderInfoBuilder(
+      name = "rawls",
+      pactSource = PactSource
+        .PactBrokerWithSelectors(pactBrokerUrl)
+        .withAuth(BasicAuth(pactBrokerUser, pactBrokerPass))
+        .withPendingPactsEnabled(ProviderTags(providerBranch))
+        .withConsumerVersionSelectors(consumerVersionSelectors)
+    )
+      .withStateManagementFunction(
+        providerStatesHandler
+//          .withBeforeEach(() => resetMocks())
```

Comment:
`resetMocks` was copied from the LeoProvider, I think, and I'm not sure if I need it - I did have some errors from it at some point and it seems to work without it.  Thoughts?

Reply:
I'm fine to leave it out until something happens that makes us add it back in!